File: flutter/task_app/lib/models/Task.dart
class Task {
  final String id;
  final String title;
  final String description;
  final String isCompleted;
  final String createdAt;

  const Task(this.id, this.title, this.description, this.isCompleted, this.createdAt);

  Map<String, dynamic> toMap() {
    return {
      '_id': id,
      'title': title,
      'description': description,
      'isCompleted': isCompleted,
      'createdAt':createdAt

    };
  }

  static Task fromMap(Map<String, dynamic> map) {
    return Task(map['id'], map['title'], map['description'], map['isCompleted'], map['createdAt']);
  }
}

File: backend/models/Task.js
import mongoose from "mongoose";

const TaskSchema = new mongoose.Schema({
  title: { type: String, Required:true },
  description: { type: String },
  isCompleted: { type: Boolean, default: false },
  userId: { type: mongoose.Types.ObjectId, Required:true },
  createdAt: { type: Date, default: new Date() },
});

const Task = mongoose.model("Task", TaskSchema);
export default Task;